conjur-api/Client.Methods.cs

```diff
@@ -24,6 +28,33 @@ public Variable Variable(string name)
             return new Variable(this, name);
         }
 
+        /// <summary>
+        /// Search for variables
+        /// </summary>
+        /// <param name="query">Query for search.</param>
+        /// <param name="actingAs">Fully-qualified Conjur ID of a role to act as.</param>
+        /// <returns>Returns IEnumerable to VariableInfo.</returns>
+        public IEnumerable<Variable> SearchVariables(string query = null, string actingAs = null)
+        {
+            uint offset = 0;
+            List<SearchVariableResult> searchVarsResult;
+            do
+            {
+                string urlWithParams = $"authz/{GetAccountName()}/resources/variable?offset={offset}"
+                                      + $"&limit={limitSearchVarListsReturned}"
+                                      + ((query != null) ? $"&search={query}" : string.Empty)
+                                      + ((actingAs != null) ? $"&acting_as={actingAs}" : string.Empty);
+
+                searchVarsResult = JsonSerializer<List<SearchVariableResult>>.Read(this.AuthenticatedRequest(urlWithParams));
+                foreach (SearchVariableResult searchVarResult in searchVarsResult)
+                {
+                    yield return new Variable(this, searchVarResult.Id);
+                }
+
+                offset += (uint)searchVarsResult.Count;
+            } while (searchVarsResult.Count > 0);
```

Comment:
Good way to hide the limit/offset. But consider documenting this so the user knows to expect network requests happening behind the scenes.